quic/s2n-quic/src/tests/zero_length_cid_client_connection_migration.rs

```diff
@@ -0,0 +1,234 @@
+        client_config.set_initial_max_data(10_000_000);
+        client_config.set_initial_max_stream_data_bidi_local(1_000_000);
+        client_config.set_initial_max_stream_data_bidi_remote(1_000_000);
+        client_config.set_disable_active_migration(false);
+        client_config.set_active_connection_id_limit(3);
+
+        // create a zero-length Source CID
+        let scid = quiche::ConnectionId::default();
+
+        let socket = handle.builder().build()?.socket();
+        let migrated_socket = handle.builder().build()?.socket();
+
+        // Create a QUIC connection and initiate handshake.
+        let conn = quiche::connect(
+            Some("localhost"),
+            &scid,
+            socket.local_addr().unwrap(),
+            server_addr,
+            &mut client_config,
+        )
+        .unwrap();
+
+        // Check if the client is using zero-length CID
+        assert_eq!(conn.source_id().len(), 0);
+
+        start_quiche_client(conn, socket, migrated_socket, server_addr).unwrap();
+
+        Ok(())
+    })
+    .unwrap();
+
+    let initial_cid_vec = initial_cid_event.lock().unwrap();
+    // The server should only perform one handshake with a successful
+    // connection mgiration. Hence, it should only receive one Initial packet
+    // with Crypto frame
+    assert_eq!(initial_cid_vec.len(), 1);
+    // Verify if the client's original CID is zero-length
+    assert_eq!(initial_cid_vec[0].len(), 0);
+
+    // Verify if the new path is validated
+    let path_challenge_statuses = path_challenge_event.lock().unwrap();
+    let path_validated = path_challenge_statuses
+        .iter()
+        .any(|status| matches!(status, events::PathChallengeStatus::Validated { .. }));
+    assert!(path_validated);
+}
+
+pub fn start_quiche_client(
+    mut client_conn: quiche::Connection,
+    socket: Socket,
+    migrated_socket: Socket,
+    server_addr: SocketAddr,
+) -> Result<()> {
+    let mut out = [0; QUICHE_MAX_DATAGRAM_SIZE];
+    let mut buf = [0; QUICHE_MAX_DATAGRAM_SIZE];
+    let application_data = "Test Migration";
+
+    primary::spawn(async move {
+        client_conn.timeout();
+
+        // Write Initial handshake packets
+        let (write, send_info) = client_conn.send(&mut out).expect("Initial send failed");
+        socket
+            .send_to(send_info.to, NotEct, out[..write].to_vec())
+            .unwrap();
+
+        let mut path_probed = false;
+        let mut req_sent = false;
+        loop {
+            // We need to check if there is a timeout event at the beginning of
+            // each loop to make sure that the connection will close properly when
+            // the test is done.
+            client_conn.on_timeout();
+            // Quiche doesn't handle IO. So we need to handle events happen
+            // on both the original socket and the migrated socket
+            let sockets = vec![&socket, &migrated_socket];
+            for active_socket in sockets {
```

Comment:
you aren't using `sockets` for anything else so you can just do:
```suggestion
            for active_socket in [&socket, &migrated_socket] {
```